modules/swaylock/hm.nix

```diff
@@ -26,6 +26,9 @@ in
     })
   ];
   options.stylix.targets.swaylock = {
+    # FIXME: autoEnable shouldn't depend on a config option, because it is used in the docs
+    # Maybe we need a function that separates the real bool from a literal text bool?
+    # Anything dynamic like this needs additional care when documenting it.
     enable =
       config.lib.stylix.mkEnableTarget "Swaylock"
         # When the state version is older than 23.05, Swaylock enables itself
```

Comment:
Looking treewide, it seems the only other cases of dynamic `autoEnable` values not caught by this PR are a handful of `config.stylix.image != null` conditions.

This can't be caught by the docs, because the `stylix.image` option _will_ exist even in the stripped down eval. However it's still problematic, because the nuance of a dynamic condition can't be reflected in the `defaultText` or `example` generated by `mkEnableTarget`.

I'd recommend either adding another helper function for creating these more advanced enable-target options, or having the handful of cases that need dynamic conditions manually implement the enable option with `lib.mkOption`; defining their own bespoke `defaultText`, `example`, etc.